server/src/main/java/org/elasticsearch/transport/TransportStats.java

```diff
@@ -36,13 +36,149 @@
     private final long rxSize;
     private final long txCount;
     private final long txSize;
+    private static final Version CONNECTION_STATS_INTRODUCED = Version.V_8_0_0;
+    private final InboundConnectionsStats inboundConnectionsStats;
 
-    public TransportStats(long serverOpen, long rxCount, long rxSize, long txCount, long txSize) {
+    public InboundConnectionsStats getInboundConnectionsStats() {
+        return inboundConnectionsStats;
+    }
+
+    public static class InboundConnectionsStats implements Writeable, ToXContentFragment {
+        private final long openedChannels;
+        private final long closedChannels;
+        private final long requestsReceivedCount;
+        private final long requestsReceivedBytes;
+        private final long responsesSentCount;
+        private final long responseSentBytes;
+        private final long keepAlivePingsReceivedCount;
+        private final long keepAlivePingsReceivedBytes;
+        private final long keepAlivePongsSentCount;
+        private final long keepAlivePongsSentBytes;
+
+        static final String INBOUND_CONNECTIONS = "inbound_connections";
```

Comment:
As suggested in the other comment and https://github.com/elastic/elasticsearch/issues/36127, probably better to make this a gauge and only keep the closed counter.